pkg/controller/utils/utils.go

```diff
@@ -65,6 +68,20 @@ func BuildClusterAPIClientFromKubeconfig(kubeconfigData string) (client.Client,
 	})
 }
 
+// TBD
+func GetClusterAPIClient(deploymentConditions []hivev1.ClusterDeploymentCondition, conditionType hivev1.ClusterDeploymentConditionType, kubeconfigData string) (client.Client, error) {
```

Comment:
What do you think about simplifying the callers and just passing in ClusterDeployment here, and assume that we're checking the unreachable condition in this function.

Reply:
Sounds good. Looking in to it.

Reply:
The issue I found when I was implementing is that, to retrieve the `kubeConfigData` I need to access the `reconcile.Reconciler` which is specific to each controller. So had to pass cluster deployment and kubeConfigData.

Reply:
Sorry I was thinking just eliminating the need for the callers to find the condition and pass the type, so i should have said just pass cluster deployment + kubeconfigdata.